src/nvim/extmark.c

```diff
@@ -394,6 +414,7 @@ void extmark_splice_delete(buf_T *buf, int l_row, colnr_T l_col, int u_row, coln
         } else {
           invalidated = true;
           mt_itr_rawkey(itr).flags |= MT_FLAG_INVALID;
+          mt_meta_invalid(buf->b_marktree, mark, true);
```

Comment:
Hmm, instead of keeping an extra ` meta_invalid` counter we should actually remove the count from `meta_root` and also any other parent node (and add them back on undo, if re-ravidated). as invalid nodes are not rendered it is safe to skip them when actually using the meta counts to filter nodes at render time.

Reply:
Yeah that's what I started implementing at first but when I realized that those counters probably propagate throughout the child/parent nodes I wasn't sure the added complexity would be worth it (if I even managed to figure out how to implement it in marktree.c). I could take another look if that's what we want to do.

Reply:
I think we should be able to reuse `marktree_revise_flags` which already does this. `meta_describe_key` needs an extra condition to exclude invalid nodes, just like we already exclude end nodes.